spec/intl.html

```diff
@@ -2148,13 +2192,24 @@ <h1>Temporal.Calendar.prototype.mergeFields ( _fields_, _additionalFields_ )</h1
             1. Perform ? RequireInternalSlot(_calendar_, [[InitializedTemporalCalendar]]).
             1. Set _fields_ to ? ToObject(_fields_).
             1. Set _additionalFields_ to ? ToObject(_additionalFields_).
-            1. If _calendar_.[[Identifier]] is *"iso8601"*, then
-              1. Return ? DefaultMergeCalendarFields(_fields_, _additionalFields_).
-            1. Let _fieldsCopy_ be OrdinaryObjectCreate(%Object.prototype%).
-            1. Perform ? CopyDataProperties(_fieldsCopy_, _fields_, « », « *undefined* »).
-            1. Let _additionalFieldsCopy_ be OrdinaryObjectCreate(%Object.prototype%).
+            1. Let _additionalFieldsCopy_ be OrdinaryObjectCreate(*null*).
             1. Perform ? CopyDataProperties(_additionalFieldsCopy_, _additionalFields_, « », « *undefined* »).
-            1. Return CalendarDateMergeFields(_calendar_.[[Identifier]], _fieldsCopy_, _additionalFieldsCopy_).
+            1. NOTE: Every property of _additionalFieldsCopy_ is a data property with non-*undefined* value, but some property keys may be Symbols.
+            1. Let _excludedKeys_ be an empty Set.
+            1. Let _additionalKeys_ be ! _additionalFieldsCopy_.[[OwnPropertyKeys]]().
+            1. For each element _key_ of _additionalKeys_, do
+              1. Add _key_ to _excludedKeys_.
+              1. If _calendar_.[[Identifier]] is *"iso8601"*, then
+                1. If _key_ is *"month"*, add *"monthCode"* to _excludedKeys_.
+                1. Else if _key_ is *"monthCode"*, add *"month"* to _excludedKeys_.
+              1. Else,
+                1. Let _toExclude_ be CalendarKeysToExclude(_calendar_, _key_).
+                1. Set _excludedKeys_ to the union of _excludedKeys_ and the set of _toExclude_'s elements.
+            1. Let _merged_ be OrdinaryObjectCreate(%Object.prototype%).
+            1. Let _excludedKeysList_ be the List of _excludedKeys_' elements.
+            1. Perform ? CopyDataProperties(_merged_, _fields_, _excludedKeysList_, « *undefined* »).
+            1. Perform ! CopyDataProperties(_merged_, _additionalFieldsCopy_, « »).
```

Comment:
I think these steps would be more comprehensible if they were holistic rather than field-by-field, and also if _excludedKeys_ were renamed. Suggestion:
```suggestion
            1. If _calendar_.[[Identifier]] is *"iso8601"*, then
              1. Let _overriddenKeys_ be ISOFieldKeysToOverride(_additionalKeys_).
            1. Else,
              1. Let _overriddenKeys_ be CalendarFieldKeysToOverride(_calendar_, _additionalKeys_).
            1. Let _merged_ be OrdinaryObjectCreate(%Object.prototype%).
            1. Perform ? CopyDataProperties(_merged_, _fields_, _overriddenKeys_, « *undefined* »).
            1. Perform ! CopyDataProperties(_merged_, _additionalFieldsCopy_, « »).
```

It would also be nice if property order aligned with `{ ...fields, ...additionalFields }` (here and in DefaultMergeCalendarFields, of course):
```suggestion
            1. If _calendar_.[[Identifier]] is *"iso8601"*, then
              1. Let _overriddenKeys_ be ISOFieldKeysToOverride(_additionalKeys_).
            1. Else,
              1. Let _overriddenKeys_ be CalendarFieldKeysToOverride(_calendar_, _additionalKeys_).
            1. NOTE: The following steps ensure that property iteration order of _merged_ matches that of _fields_ as modified by omitting overridden keys and appending non-overlapping keys from _additionalFields_ in iteration order.
            1. Let _fieldsKeys_ be ! _fields_.[[OwnPropertyKeys]]().
            1. For each element _key_ of _fieldsKeys_, do
              1. Let _propValue_ be *undefined*.
              1. If _overriddenKeys_ contains _key_,
                1. Set _propValue_ to ! Get(_additionalFieldsCopy_, _key_).
              1. Else,
                1. Let _desc_ be ? <emu-meta effects="user-code">_fields_.[[GetOwnProperty]]</emu-meta>(_key_).
                1. If _desc_ is not *undefined* and _desc_.[[Enumerable]] is *true*, then
                  1. Set _propValue_ to ? Get(_from_, _key_).
              1. If _propValue_ is not *undefined*, perform ! CreateDataPropertyOrThrow(_merged_, _key_, _propValue_).
            1. Perform ! CopyDataProperties(_merged_, _additionalFieldsCopy_, « »).
```

Reply:
I've taken a look at how this would shake out. I could go either way. I think iterating and calling the implementation-defined operation once per key is probably closer to how it'd be implemented in an engine, and codifies the fact that keys must exclude themselves. On the other hand, calling the implementation-defined operation on the set of keys.

I don't feel strongly about matching `{ ...fields, ...additionalFields }` either — if the returned object has a null prototype it's not possible to polyfill it that way anyway.

Reply:
I believe you are mistaken, because the third argument in `CopyDataProperties(_merged_, _fieldsCopy_, _overriddenKeys_, « *undefined* »)` has the effect of omitting keys that are present in _fieldsCopy_ which the subsequent `CopyDataProperties(_merged_, _additionalFieldsCopy_, « »)` can add back in a different position. For a simple example, consider how this algorithm requires `Reflect.ownKeys(Temporal.Calendar.from("iso8601").mergeFields({ year: 2023, month: 2, day: 18 }, { month: 3 }))` to return `["year", "day", "month"]` rather than `["year", "month", "day"]`.

Reply:
#2504